src/middleware/validator/assignments.validator.ts

```diff
@@ -0,0 +1,12 @@
+import { check } from 'express-validator'
+
+import validate from './generic.validator'
+
+const name = check('name').isString().trim().isLength({ max: 128 })
+const grading_type = check('grading_type').isString().trim().isLength({ max: 128 })
+const category_name = check('category_name').isString().trim().isLength({ max: 128 })
+const description = check('description').isString().trim().isLength({ max: 128 })
```

Comment:
Description should likely not have a max as it's an input for Profs to input paragraphs of details about an assignment.